### ISSUE TITLE
"Include Today" on date filter does not work for SQL queries

### ISSUE THREAD

**Issue by MarkRx** [metabase/metabase#7339]
The "include today" button on a date field filter is not working. There are two bugs:

1. The data is not being filtered properly (today is not being included). This is because a trunc is being performed when it should not be (https://github.com/metabase/metabase/issues/7339#issuecomment-380867574)
2. The UI is not displaying the correct text (It still says "Yesterday" when "include today" is checked)

# Example:

**Previous 1 days (without include today):**
Results:
1. Wednesday, April 11, 2018 11:57 AM

**Previous 1 days (with include today checked):**
Results (this is wrong):
Wednesday, April 11, 2018 11:57 AM

UI (this is wrong):
![image](https://user-images.githubusercontent.com/17504314/38689974-fbba4ea6-3e4a-11e8-8f77-dd73f0b6c8b5.png)

**POST http://localhost:3000/api/dataset:**
Observe that include today is properly adding a ~ at the end.
```
{
	"database": 2,
	"type": "native",
	"native": {
		"query": "select *\nfrom mbase.DATE_TEST\nwhere {{mydate}}",
		"collection": "APEX_APPL_PLUGIN_ATTR_VALUES",
		"template_tags": {
			"mydate": {
				"id": "ecca0929-c107-dc63-a104-26a420b7c8af",
				"name": "mydate",
				"display_name": "Mydate",
				"type": "dimension",
				"dimension": ["field-id", 3745],
				"widget_type": "date/all-options"
			}
		}
	},
	"parameters": [{
			"type": "date/all-options",
			"target": ["dimension", ["template-tag", "mydate"]],
			"value": "past1days~"
		}
	]
}
```

**Response:**
```
{
	"started_at": "2018-04-12T16:12:13.611Z",
	"json_query": {
		"database": 2,
		"type": "native",
		"native": {
			"query": "select *\nfrom mbase.DATE_TEST\nwhere {{mydate}}",
			"collection": "APEX_APPL_PLUGIN_ATTR_VALUES",
			"template_tags": {
				"mydate": {
					"id": "ecca0929-c107-dc63-a104-26a420b7c8af",
					"name": "mydate",
					"display_name": "Mydate",
					"type": "dimension",
					"dimension": ["field-id", 3745],
					"widget_type": "date/all-options"
				}
			}
		},
		"parameters": [{
				"type": "date/all-options",
				"target": ["dimension", ["template-tag", "mydate"]],
				"value": "past1days~"
			}
		],
		"constraints": {
			"max-results": 10000,
			"max-results-bare-rows": 2000
		}
	},
	"average_execution_time": null,
	"status": "completed",
	"context": "ad-hoc",
	"row_count": 1,
	"running_time": 8,
	"data": {
		"rows": [[1, "2018-04-11T11:57:34.000-04:00"]],
		"columns": ["ID", "DT"],
		"native_form": {
			"query": "select *\nfrom mbase.DATE_TEST\nwhere trunc(\"MBASE\".\"DATE_TEST\".\"DT\", 'dd') = ?",
			"collection": "APEX_APPL_PLUGIN_ATTR_VALUES",
			"template_tags": {
				"mydate": {
					"id": "ecca0929-c107-dc63-a104-26a420b7c8af",
					"name": "mydate",
					"display_name": "Mydate",
					"type": "dimension",
					"dimension": ["field-id", 3745],
					"widget_type": "date/all-options"
				}
			},
			"params": ["2018-04-11T04:00:00.000Z"]
		},
		"cols": [{
				"name": "ID",
				"display_name": "ID",
				"base_type": "type/Decimal",
				"remapped_to": null,
				"remapped_from": null
			}, {
				"name": "DT",
				"display_name": "Dt",
				"base_type": "type/DateTime",
				"remapped_to": null,
				"remapped_from": null
			}
		],
		"results_metadata": {
			"checksum": "tWqGN0Z0shz/hBnxx1k39A==",
			"columns": [{
					"base_type": "type/Decimal",
					"display_name": "ID",
					"name": "ID"
				}, {
					"base_type": "type/DateTime",
					"display_name": "Dt",
					"name": "DT"
				}
			]
		}
	}
}
```

**Question:**
> select *
> from mbase.DATE_TEST
> where {{mydate}}

mydate is a field filter of type "date filter" mapped to MBASE > Date Test > dt

**SQL Setup (example data created 4/12/2018 11:57 AM):**
> create table date_test (
>     id NUMBER PRIMARY KEY,
>     dt TIMESTAMP(3)
> );
> 
> select sysdate - 1 from dual;
> 
> insert into date_test values(1, sysdate - 1);
> insert into date_test values(2, sysdate);
> insert into date_test values(3, sysdate + 1);

Browser: Chrome 64.0.3282.119
OS: Windows 7
Databases: Oracle 11g
Version: 0.29.0-SNAPSHOT c61a31f
Host Env: Windows 7
Internal Database: H2

**Comment by MarkRx** [comment]
Looks like it is because a trunc is being performed. From v$sql:

> -- Metabase:: userID: 1 queryType: native queryHash: fb739d4d78e776d975e48425e7882b3e6b07ca0e0be23ce1942891c764f426a8 select * from mbase.DATE_TEST where trunc("MBASE"."DATE_TEST"."DT", 'dd') = :1

**Comment by MarkRx** [comment]
It does not work for SQL queries but works for UI queries.

The UI calls https://github.com/metabase/metabase/blob/master/src/metabase/query_processor/middleware/parameters/dates.clj#L218
whereas sql calls https://github.com/metabase/metabase/blob/master/src/metabase/query_processor/middleware/parameters/dates.clj#L201 which strips out the special information that indicates that today should be included

**Comment by MarkRx** [comment]
@cgopalan is working on this

**Comment by MarkRx** [comment]
PR #7390

**Comment by salsakran** [comment]
fixed in #7390
